x/leverage/keeper/grpc_query.go

```diff
@@ -414,7 +414,12 @@ func (q Querier) MaxWithdraw(
 		// will be nil and the resulting value will be what
 		// can safely be withdrawn even with missing prices.
 		// On non-nil error here, max withdraw is zero.
-		uToken, _, err := q.userMaxWithdraw(ctx, addr, denom)
+		uToken, _, err := q.Keeper.userMaxWithdraw(ctx, addr, denom)
+		if err == nil {
+			var moduleMaxWithdraw sdk.Int
+			moduleMaxWithdraw, err = q.Keeper.ModuleMaxWithdraw(ctx, uToken)
+			uToken.Amount = sdk.MinInt(uToken.Amount, moduleMaxWithdraw)
+		}
```

Comment:
Suggested error edit is (probably not correct) because it would skip the "nonOracleError' logic below.

Reply:
I changed the code in the second commit. Now the 'NonOracleError' flow is more explicit. Previously it was in roundabout way at the end because there was fewer code paths that allowed it to be that way.